### ISSUE TITLE
NixOS-kFreeBSD

### ISSUE THREAD

**Issue by moretea** [nix-community/google-summer-of-code#20]

**Comment by moretea** [comment]
@AndersonTorres would you be willing to mentor this?

**Comment by AndersonTorres** [comment]
I'm a bit scaried about it. Is being amentor a huge rescponsibility for a, what to say, amateur programmer like me?

**Comment by zimbatm** [comment]
@AndersonTorres as a mentor you would be responsible to help out the student to carry out his tasks. It means bringing your own experience on development and communicate with the student to help him out break down the project into more manageable chunks.

One thing to consider with these proposals is whenever the work would fit in the span of 3 months. Divided by 3 because they are students and still learning to code, plus there is a communication overhead.

**Comment by AndersonTorres** [comment]
@zimbatm @moretea 

I was thinking with my black cat.

Even discounting my low self-esteem, I don't consider myself a skilled programmer; in fact I came back to programming because of NixOS :). Add to it the fact I have no formal education (in terms of a undergrad degree).

NixOS-kFreeBSD is huge and ambitious. We are trying to build a whole system almost from scratch here! As far as I don't know the full GSOC rules, maybe a less ambitious task can be made, as "compile and run Nix on an already installed FreeBSD environment" with some small Nixpkgs testing subset.

Maybe the only other problem is time. But today I feel compelled to mentor this little project. There is some deadline we need to meet for now?

**Comment by zimbatm** [comment]
Agreed, the scope for the project should be reduced. A lot. Just going through the bootstrap stdenv would already be quite an achievement for a student I believe.
